declarative/loginmanager/usersmodel.cpp

```diff
@@ -0,0 +1,91 @@
+ *
+ * $BEGIN_LICENSE:GPL3+$
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * $END_LICENSE$
+ ***************************************************************************/
+
+#include <QtCore/QDir>
+#include <QtCore/QUrl>
+#include <QtCore/QSettings>
+#include <QDebug>
+
+#include <QtAccountsService/UserAccount>
+
+#include "usersmodel.h"
+
+UsersModel::UsersModel(QObject *parent)
+    : QtAccountsService::UsersModel(parent)
+{
+}
+
+QHash<int, QByteArray> UsersModel::roleNames() const
+{
+    QHash<int, QByteArray> roles = QtAccountsService::UsersModel::roleNames();
+    roles.insert(BackgroundModeRole, QByteArrayLiteral("backgroundMode"));
+    roles.insert(PictureUrlRole, QByteArrayLiteral("pictureUrl"));
+    roles.insert(FillModeRole, QByteArrayLiteral("fillMode"));
+    roles.insert(PrimaryColorRole, QByteArrayLiteral("primaryColor"));
+    roles.insert(SecondaryColorRole, QByteArrayLiteral("secondaryColor"));
+    return roles;
+}
+
+QVariant UsersModel::data(const QModelIndex &index, int role) const
+{
+    if (!index.isValid())
+        return QVariant();
+
+    if (role < BackgroundModeRole)
+        return QtAccountsService::UsersModel::data(index, role);
+
+    QtAccountsService::UserAccount *user = userAccount(index);
+
+    const QString confFileName = QDir::home().absoluteFilePath(
+                QStringLiteral(".local/share/sddm/backgrounds/%1.conf").arg(user->userName()));
+    const QDir backgroundDir = QDir::home().absoluteFilePath(QLatin1String(".local/share/sddm/backgrounds"));
+
+    // Read the configuration even if the file doesn't exist,
+    // in that case we'll return default values to avoid QML errors
+    // of assigning null or undefined values to color properties
+    QSettings conf(confFileName, QSettings::IniFormat);
+    conf.beginGroup(QLatin1String("Background"));
+
+    switch (role) {
+    case BackgroundModeRole:
+        return conf.value(QLatin1String("Mode"), QString());
+    case PictureUrlRole: {
+        const QString fileName = conf.value(QLatin1String("PictureUrl"), QString()).toString();
+        if (fileName.isEmpty())
+            return QUrl();
+        return QUrl::fromLocalFile(backgroundDir.absoluteFilePath(fileName));
+    }
+    case FillModeRole:
+        return conf.value(QLatin1String("FillMode"), QLatin1String("pad"));
```

Comment:
`preserve-aspect-crop` seems like a nicer default. I think that's what GNOME, KDE, and macOS use by default.